librz/arch/p/analysis/analysis_x86_zydis.c

```diff
@@ -0,0 +1,3778 @@
+	case ZYDIS_MNEMONIC_SUBPS:
+	case ZYDIS_MNEMONIC_SUBSS:
+	case ZYDIS_MNEMONIC_SQRTPS:
+	case ZYDIS_MNEMONIC_SQRTSS:
+	// SSE2 Instructions:
+	// case ZYDIS_MNEMONIC_ADDPS:
+	// case ZYDIS_MNEMONIC_ADDSS:
+	// case ZYDIS_MNEMONIC_ADDPD:
+	// case ZYDIS_MNEMONIC_SUBPS:
+	// case ZYDIS_MNEMONIC_SUBSS:
+	// case ZYDIS_MNEMONIC_MULPS:
+	// case ZYDIS_MNEMONIC_MULSS:
+	// case ZYDIS_MNEMONIC_DIVPS:
+	// case ZYDIS_MNEMONIC_DIVSS:
+	// case ZYDIS_MNEMONIC_MAXPS:
+	// case ZYDIS_MNEMONIC_MAXSS:
+	// case ZYDIS_MNEMONIC_MINPS:
+	// case ZYDIS_MNEMONIC_MINSS:
+	// case ZYDIS_MNEMONIC_CMPPS:
+	// case ZYDIS_MNEMONIC_CMPSS:
+	case ZYDIS_MNEMONIC_CVTPI2PS:
+	case ZYDIS_MNEMONIC_CVTSS2SI:
+	case ZYDIS_MNEMONIC_CVTPS2PI:
+	case ZYDIS_MNEMONIC_CVTSI2SS:
+	case ZYDIS_MNEMONIC_CVTTPD2DQ:
+	case ZYDIS_MNEMONIC_CVTPD2PS:
+	case ZYDIS_MNEMONIC_CVTPS2PD:
+	case ZYDIS_MNEMONIC_CVTSD2SI:
+	case ZYDIS_MNEMONIC_CVTSS2SD:
+	case ZYDIS_MNEMONIC_CVTSD2SS:
+	case ZYDIS_MNEMONIC_MOVAPD:
+	case ZYDIS_MNEMONIC_MOVAPS:
+	case ZYDIS_MNEMONIC_MOVD:
+	case ZYDIS_MNEMONIC_MOVQ:
+	case ZYDIS_MNEMONIC_PMOVMSKB:
+	// case ZYDIS_MNEMONIC_PSHUFPS:
+	case ZYDIS_MNEMONIC_PSHUFD:
+	case ZYDIS_MNEMONIC_PSHUFHW:
+	case ZYDIS_MNEMONIC_PSHUFLW:
+	case ZYDIS_MNEMONIC_PTEST:
+	// SSE3 Instructions:
+	case ZYDIS_MNEMONIC_ADDPD:
+	// case ZYDIS_MNEMONIC_ADDPS:
+	// case ZYDIS_MNEMONIC_ANDPD:
+	// case ZYDIS_MNEMONIC_ANDPS:
+	case ZYDIS_MNEMONIC_CMPPD:
+	// case ZYDIS_MNEMONIC_CMPPS:
+	case ZYDIS_MNEMONIC_CVTDQ2PS:
+	case ZYDIS_MNEMONIC_CVTPS2DQ:
+	// case ZYDIS_MNEMONIC_CVTSS2SD:
+	// case ZYDIS_MNEMONIC_CVTSD2SS:
+	// case ZYDIS_MNEMONIC_CVTTPD2DQ:
+	// case ZYDIS_MNEMONIC_CVTPD2PS:
+	// case ZYDIS_MNEMONIC_MOVAPS:
+	case ZYDIS_MNEMONIC_MOVDDUP:
+	case ZYDIS_MNEMONIC_MOVSS:
+	// case ZYDIS_MNEMONIC_PSHUFPS:
+	case ZYDIS_MNEMONIC_RCPPS:
+		// case ZYDIS_MNEMONIC_RSQRTPS:
+		op->family = RZ_ANALYSIS_OP_FAMILY_SSE;
+		break;
+	default: break;
+	}
+}
+
+static inline ZydisMachineMode select_mode(RzAnalysis *a) {
+	switch (a->bits) {
+	case 64:
+		return ZYDIS_MACHINE_MODE_LONG_64;
+	case 32:
+		return ZYDIS_MACHINE_MODE_LONG_COMPAT_32;
+	case 16:
+		return ZYDIS_MACHINE_MODE_LONG_COMPAT_32;
+	default:
+		return 0;
+	}
+}
+
+static int analyze_op(RzAnalysis *a, RzAnalysisOp *op, ut64 addr, const ut8 *buf, int len, RzAnalysisOpMask mask) {
+	X86ZYDISContext *zydx = (X86ZYDISContext *)a->plugin_data;
+
+	ZydisMachineMode mode = select_mode(a);
+	ZydisStackWidth st_mode;
+	int n, ret;
+
+	zydx->omode = mode;
+	switch (mode) {
+	case ZYDIS_MACHINE_MODE_LONG_64:
+		st_mode = ZYDIS_STACK_WIDTH_64;
+		break;
+	case ZYDIS_MACHINE_MODE_LONG_COMPAT_16:
+		st_mode = ZYDIS_STACK_WIDTH_16;
+		break;
+	default:
+		st_mode = ZYDIS_STACK_WIDTH_32;
+		break;
+	}
+
+	ZydisDecoder decoder;
+	ret = ZydisDecoderInit(&decoder, mode, st_mode);
+	if (!ZYAN_SUCCESS(ret)) {
+		return 0;
+	}
+	zydx->zydecode = RZ_NEW0(ZydisDecodedInstruction);
+	zydx->zydeop = RZ_NEWS(ZydisDecodedOperand, ZYDIS_MAX_OPERAND_COUNT);
```

Comment:
It seems like you free them before returning. Please don't allocate them on the heap then. Just allocate them on the stack and assign the pointer here. You can set them to `NULL` before returning.